### PR TITLE
[OTE-882] Add prefix to accountplus keeper

### PR DESCRIPTION
**Pull request by jerryfan01234**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Release Notes

- **New Features**
  - Introduced a migration process for account states to transition to a prefixed key structure.
  - Added a new constant for account state key prefixes to enhance data organization.

- **Bug Fixes**
  - Improved error handling and data retrieval methods for account states, ensuring consistency and reliability.

- **Tests**
  - Added comprehensive tests for the migration process and account state management, ensuring functionality and correctness.

### CODE REVIEW

File: protocol/x/accountplus/keeper/keeper.go [hunk 1]
```diff
@@ -144,7 +147,7 @@ func (k Keeper) GetAllAuthenticatorData(ctx sdk.Context) ([]types.AuthenticatorD
 	return accountAuthenticators, nil
 }
 
-func GetAccountPlusStateWithTimestampNonceDetails(
+func AccountStateFromTimestampNonceDetails(
```
Comment (teddyding) [inline protocol/x/accountplus/keeper/keeper.go:150]: Nit: this is only used for newly initialized account state
```suggestion
func NewAccountStateFromTimestampNonceDetails(
```

File: protocol/x/accountplus/keeper/keeper.go [hunk 2]
```diff
@@ -46,26 +53,22 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 func (k Keeper) InitializeForGenesis(ctx sdk.Context) {
 }
 
-// Get all account details pairs in store
+// Get all AccountStates from kvstore
 func (k Keeper) GetAllAccountStates(ctx sdk.Context) ([]types.AccountState, error) {
 	store := ctx.KVStore(k.storeKey)
```
Comment (teddyding) [inline protocol/x/accountplus/keeper/keeper.go:58]: Simplify next 3 lines by:

```
iterator := storetypes.KVStorePrefixIterator(ctx.KVStore(k.storeKey), []byte(types.AccountStateKeyPrefix))
```

File: protocol/x/accountplus/keeper/keeper_test.go [hunk 3]
```diff
@@ -53,6 +53,52 @@ func (s *KeeperTestSuite) SetupTest() {
...
+	)
+
+	// GetAccountState
+	_, found := s.tApp.App.AccountPlusKeeper.GetAccountState(ctx, sdk.AccAddress([]byte(address)))
+	s.Require().True(found, "Account state not found")
+
+	// GetAllAccountStates
+	accountStates, err := s.tApp.App.AccountPlusKeeper.GetAllAccountStates(ctx)
+	s.Require().NoError(err, "Should not error when getting all account states")
+	s.Require().Equal(len(accountStates), 1, "Incorrect number of AccountStates retrieved")
```
Comment (teddyding) [inline protocol/x/accountplus/keeper/keeper_test.go:80]: Here and below let's also compare returned object(s)

File: protocol/app/upgrades/v8.0/upgrade_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+)
+
+type UpgradeTestSuite struct {
+	suite.Suite
+
+	tApp *testapp.TestApp
+	Ctx  sdk.Context
+}
+
+func TestUpgradeTestSuite(t *testing.T) {
```
Comment (teddyding) [inline protocol/app/upgrades/v8.0/upgrade_test.go:20]: For my understanding did you write these tests based on some existing template? Have we considered using the conventional upgrade containter test ([example](https://github.com/dydxprotocol/v4-chain/blob/ebc4a8efa93cb2dffe6734c1915829c889c51e9d/protocol/app/upgrades/v7.0/upgrade_container_test.go#L29)), which is more end-to-end?

The current test more like a unit test on `migrateAccountplusAccountState` - no upgrade is actually being run and we are just calling `migrateAccountplusAccountState` with whitebox set-up and postcheck. Probably more accurate to rename this test to `TestMigrateAccountplusAccountState`.

File: protocol/app/upgrades/v8.0/upgrade_test.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	// Verify that unprefixed keys are deleted and prefixed keys exist
+	for _, addr := range addresses {
+		accAddress := sdk.AccAddress([]byte(addr))
+
+		// Check that the old key no longer exists
+		bzOld := store.Get(accAddress)
+		s.Require().Nil(bzOld, "Unprefixed AccountState should be deleted for %s", addr)
+
+		// Check that the new prefixed key exists
+		bzNew := prefixStore.Get(accAddress)
```
Comment (teddyding) [inline protocol/app/upgrades/v8.0/upgrade_test.go:68]: Should be relatively easy to not only check the node keys but also the content

File: protocol/app/upgrades/v8.0/upgrade.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+		}
+
+		value := iterator.Value()
+		var accountState accountplustypes.AccountState
+		if err := k.GetCdc().Unmarshal(value, &accountState); err != nil {
+			panic(fmt.Sprintf("failed to unmarshal AccountState for key %X: %s", key, err))
+		}
+
+		// SetAccountState stores with prefix
+		k.SetAccountState(ctx, key, accountState)
```
[the remark this answers is not in the document]
Comment (jerryfan01234) [reply protocol/app/upgrades/v8.0/upgrade.go:38]: To be safe, I will collect the items and set and remove outside of iteration

**Review by teddyding — APPROVED** [review]

**Pull request merged**